ksql-rest-app/src/test/java/io/confluent/ksql/api/AuthTest.java

```diff
@@ -137,6 +175,60 @@ public void shouldFailInsertRequestWithIncorrectRole() throws Exception {
     shouldFailInsertRequest(USER_WITH_INCORRECT_ROLE, USER_WITH_INCORRECT_ROLE_PWD);
   }
 
+  @Test
+  public void shouldExecutePullQueryWithApiSecurityContext() throws Exception {
```

Comment:
Interesting that these three tests are essentially duplicates of `super.shouldExecutePullQuery();` (and similar) but I can't think of an easy way to get the last assertion in there without the duplication.

Happy to leave this as is, just thinking -out loud- on a screen.

Reply:
Yeah, I'm not super happy about it either. I think we need to think about how we can tidy up / find a better way of reusing tests in different contexts.
Another possibility is we extract the tests we want to reuse into a different class and call it directly from the concrete test class - i.e. use composition instead of relying on inheritance.